### ISSUE TITLE
Transformer error for almost empty files

### ISSUE THREAD

**Issue by tbosch** [angular/angular#2380]
Right now, we get the following errors during our build. I think this is because we have almost empty files inside of `web/e2e_test` that look like this:

E.g. compiler_perf.dart
```
library benchmarks.e2e_test.compiler_perf;
main() {}
```

The error in the build is as follows:

```
[21:01:16] Starting 'build/pubbuild.dart'...
Loading source assets... 
Loading angular2, di/module_transformer, observe and smoke/src/default_transformer transformers... (1.5s) 
Loading angular transformers... (1.4s) 
Building benchmarks_external...
[Warning from InjectorGenerator on benchmarks_external|web/e2e_test/compiler_perf.dart]:
Unable to resolve injectable annotation angular.core.annotation_src.Decorator
[Warning from InjectorGenerator on benchmarks_external|web/e2e_test/compiler_perf.dart]:
Unable to resolve injectable annotation angular.core.annotation_src.Component
[Warning from InjectorGenerator on benchmarks_external|web/e2e_test/compiler_perf.dart]:
Unable to resolve injectable annotation angular.core.annotation_src.Formatter
[Warning from InjectorGenerator on benchmarks_external|web/e2e_test/compiler_perf.dart]:
Unable to resolve injected type name perf_api.Profiler
Build error:
Transform InjectorGenerator on benchmarks_external|web/e2e_test/compiler_perf.dart threw error: Bad state: No element
dart:collection/iterable.dart 329                      IterableBase.last
package:di/transformer/injector_generator.dart 368:30  _Processor._editMain
package:di/transformer/injector_generator.dart 72:5    _Processor.process
package:di/transformer/injector_generator.dart 25:52   InjectorGenerator.applyResolver
package:code_transformers/src/resolvers.dart 111:31    Transformer&ResolverTransformer.applyToEntryPoints.<fn>.<fn>
dart:async/future.dart 118                             Future.Future.<fn>
dart:async-patch/timer_patch.dart 16                   Timer._createTimer.<fn>
dart:isolate-patch/timer_impl.dart 385                 _Timer._runTimers
dart:isolate-patch/timer_impl.dart 411                 _Timer._handleMessage
dart:isolate-patch/isolate_patch.dart 142              _RawReceivePortImpl._handleMessage
dart:collection                                        IterableBase.last
package:di/transformer/injector_generator.dart 368:30  _Processor._editMain
package:di/transformer/injector_generator.dart 72:5    _Processor.process
package:di/transformer/injector_generator.dart 25:52   InjectorGenerator.applyResolver
package:code_transformers/src/resolvers.dart 111:31    Transformer&ResolverTransformer.applyToEntryPoints.<fn>.<fn>
dart:isolate                                           _RawReceivePortImpl._handleMessage
```

**Comment by tbosch** [comment]
Note: This does not fail the build

**Comment by kegluneq** [comment]
Note that the errors are coming from the `package:di` transformer (that is, the angular.dart transformer), not the angular 2 transformer.

Perhaps this issue should be in https://github.com/angular/di.dart?

**Comment by vicb** [comment]
will be fixed by #2505
